### PR TITLE
bpo-44110: Improve string's __getitem__ error message

### PR DESCRIPTION
**Pull request by miguendes**
https://bugs.python.org/issue44110

### CODE REVIEW

File: Lib/test/string_tests.py [hunk 1]
```diff
@@ -80,12 +80,14 @@ class subtype(self.__class__.type2test):
                 self.assertIsNot(obj, realresult)
 
     # check that obj.method(*args) raises exc
-    def checkraises(self, exc, obj, methodname, *args):
+    def checkraises(self, exc, obj, methodname, *args, **kwargs):
```
Comment (JelleZijlstra) [inline Lib/test/string_tests.py]: You can just put `expected_msg=None` here I believe.

Comment (miguendes) [reply Lib/test/string_tests.py]: Done! That's better indeed.

**Review by serhiy-storchaka — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Comment by akulakov** [comment]
A similar enhancement to list IndexError msg was recently rejected for performance reasons. I wonder if the same objection may apply here (most likely less so because the same pattern of catching index errors in a loop would be used rarely with strings). I will try to check performance later today, though I was having some odd issues with pyperf (but this may be a good chance to debug them :) ). 

related issue:
https://bugs.python.org/issue44166

**Comment by akulakov** [comment]
After looking at the pattern that was an issue with the list IndexError, it should not apply to strings, so please ignore my comment!

**Comment by serhiy-storchaka** [comment]
Yes, because it is a TypeError, not IndexError.

**Pull request merged**